tools/ci_credentials/ci_credentials/secrets_loader.py

```diff
@@ -1,16 +1,22 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
 import base64
 import json
+import re
 from json.decoder import JSONDecodeError
-from typing import Mapping, Any, Tuple, ClassVar
 from pathlib import Path
-from ci_common_utils import GoogleApi
-from ci_common_utils import Logger
+from typing import Any, ClassVar, Mapping, Tuple
+
+from ci_common_utils import GoogleApi, Logger
 
 DEFAULT_SECRET_FILE = "config"
 DEFAULT_SECRET_FILE_WITH_EXT = DEFAULT_SECRET_FILE + ".json"
 
 GSM_SCOPES = ("https://www.googleapis.com/auth/cloud-platform",)
 
+MASK_KEY_PATTERNS = ["password", "host", "user", "_key", "token", "secret", "bucket"]
```

Comment:
have any more keywords or regexes?

Reply:
can't we just play it safe and mask everything? Airbyters have access to these secrets anyways.

Reply:
the issue is if a config value is `true` or `0` or whatever, it will mask that. So I went this way. Pretty open to other ideas, if you have them, though.

Reply:
here is every secret name we have in all known connectors, if I got this code right

```
{'appid', 'account_id', 'user_auth_key', 'storage_access_key', 'client_secret', 'lwa_client_secret', 'access_key', 'credentials_json', 'survey_id', 'secret', 'refresh_token', 'site_api_key', 'aws_access_key_id', 'lwa_app_id', 'apiId', 'client_id', 'developer_token', 'app_api_key', 'base_url', 'key', 'aws_access_key', 'realm_id', 'private_key', 'aws_secret_key', 'auth_token', 'aws_secret_access_key', 'domain_url', 'account_sid', 'survey_group_id', 'role_arn', 'secret_key', 'password', 'api_token', 'application_secret', 'jwt', 'token', 'private_token', 'apiKey', 'tenant_id', 'api_secret', 'api_key', 'access_token', 'aws_key_id'}
```

generated using

```
import yaml
yamls = open("airbyte-config/init/src/main/resources/seed/source_specs.yaml", "r").read()
f=yaml.safe_load(yamls)

secrets=set()

def get_secrets(obj):
    for k,v in obj.items():
        if v.get('type', None) == 'object':
            get_secrets(v.get("properties", {}))
        elif v.get('airbyte_secret', False):
            secrets.add(k)


for spec in f:
    props=spec['spec']['connectionSpecification']['properties']
    get_secrets(props)

print(secrets)
```

Reply:
I believe these are all the known secret names we have:

```
{'access_token', 'role_arn', 'lwa_app_id', 'access_key', 'client_secret', 'password', 'service_account_info', 'account_id', 'apiId', 'api_secret', 'storage_access_key', 'token', 'api_token', 'app_id', 'jwt', 'secret_key', 'domain_url', 'client_id', 'app_api_key', 'tenant_id', 'private_key', 'base_url', 'key', 'credentials_json', 'aws_secret_key', 'secret', 'account_sid', 'user_auth_key', 'aws_key_id', 'site_api_key', 'application_secret', 'api_key', 'developer_token', 'survey_id', 'lwa_client_secret', 'private_token', 'aws_access_key_id', 'api_password', 'survey_group_id', 'auth_token', 'apiKey', 'appid', 'aws_secret_access_key', 'realm_id', 'aws_access_key', 'personal_access_token', 'apikey', 'refresh_token'}
```

It was generated using
```python
import yaml
yamls = open("airbyte-config/init/src/main/resources/seed/source_specs.yaml", "r").read()
f=yaml.safe_load(yamls)

secrets=set()

def get_secrets(obj):
    for k,v in obj.items():
        if "oneOf" in v:
            for thing in v["oneOf"]:
                if thing.get('type', None) == 'object':
                    get_secrets(thing.get('properties', {}))
        elif v.get('type', None) == 'object':
            get_secrets(v.get("properties", {}))
        elif v.get('airbyte_secret', False):
            secrets.add(k)


for spec in f:
    props=spec['spec']['connectionSpecification']['properties']
    get_secrets(props)

print(secrets)
```